omp/test/matrix/dense_kernels.cpp

```diff
@@ -75,6 +78,16 @@ class Dense : public ::testing::Test {
             std::normal_distribution<>(0.0, 1.0), rand_engine, ref);
     }
 
+    template <typename MtxType>
+    std::unique_ptr<MtxType> gen_mtx2(int num_rows, int num_cols,
```

Comment:
You do not have to call it `gen_mtx2` in C++. You can keep `gen_mtx` because you have different parameters so there is no ambiguity.